### PR TITLE
chore: Creates schema for the new mongodbatlas_resource_policy data sources

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

Creates schema for the new mongodbatlas_resource_policy data sources

Link to any related issue(s): CLOUDP-272009

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/resourcepolicy/data_source_schema.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+					},
+				},
+			},
+		},
+		"version": schema.StringAttribute{
+			Description: "A string that identifies the version of the atlas resource policy.",
+			Computed:    true,
+		},
+	}
+}
```
Comment (EspenAlbert) [inline internal/service/resourcepolicy/data_source_schema.go:102]: Notice: Plan on reusing struct models from #2548 
```
type TFResourcePolicyModel struct {
	CreatedByUser     types.Object `tfsdk:"created_by_user"`
	CreatedDate       types.String `tfsdk:"created_date"`
	ID                types.String `tfsdk:"id"`
	LastUpdatedByUser types.Object `tfsdk:"last_updated_by_user"`
	LastUpdatedDate   types.String `tfsdk:"last_updated_date"`
	Name              types.String `tfsdk:"name"`
	OrgID             types.String `tfsdk:"org_id"`
	Policies          types.List   `tfsdk:"policies"`
	Version           types.String `tfsdk:"version"`
}
```

File: internal/service/cloudprovideraccess/resource_cloud_provider_access_authorization.go [hunk 2]
```diff
@@ -260,7 +260,7 @@ func resourceCloudProviderAccessAuthorizationStateUpgradeV0(ctx context.Context,
 }
 
 func authorizeRole(ctx context.Context, client *admin.APIClient, d *schema.ResourceData, projectID string, targetRole *admin.CloudProviderAccessRole) diag.Diagnostics {
-	req := &admin.CloudProviderAccessRole{
```
Comment (EspenAlbert) [inline internal/service/cloudprovideraccess/resource_cloud_provider_access_authorization.go:263]: Changes to compile successfully.
Failing CI tests are due to this group being triggered.

File: internal/service/resourcepolicy/data_source_schema.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+func dataSourceSchema(isPlural bool) map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"created_by_user": schema.SingleNestedAttribute{
+			Description: "The user that last updated the atlas resource policy.",
+			Computed:    true,
+			Attributes: map[string]schema.Attribute{
+				"id": schema.StringAttribute{
```
Comment (lantoli) [inline internal/service/resourcepolicy/data_source_schema.go:22]: aren't we defining MarkdownDescription as well?

File: internal/service/resourcepolicy/data_source_schema.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+		"last_updated_date": schema.StringAttribute{
+			Description: "Date and time in UTC when the atlas resource policy was last updated.",
+			Computed:    true,
+		},
+		"name": schema.StringAttribute{
+			Description: "Human-readable label that describes the atlas resource policy.",
+			Computed:    true,
+		},
+		"org_id": schema.StringAttribute{
+			Required:            !isPlural,
```
Comment (lantoli) [inline internal/service/resourcepolicy/data_source_schema.go:73]: interesting

**Review by lantoli — APPROVED** [review]

File: internal/service/resourcepolicy/data_source_schema.go (hunk 3 above)
Comment (EspenAlbert) [reply internal/service/resourcepolicy/data_source_schema.go:22]: Added in [e1e988f](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2563/commits/e1e988fc2a97f90ef85314871ae78d3f1ffaf622).
Hopefully, one day we only need one `Description` 😅

**Pull request merged**